xc/device.ml

```diff
@@ -1989,43 +1981,42 @@ module Backend = struct
 
     (** Implementation of the Vbd functions that use the dispatcher for the qemu-upstream-compat backend *)
     module Vbd = struct
+
       let qemu_media_change ~xs device _type params =
         Vbd_Common.qemu_media_change ~xs device _type params;
-        let open Qmp in
-        let cd = "ide1-cd1" in
+        let cd    = "ide1-cd1" in
+        let domid = device.frontend.domid in
         if params = "" then
-          let qmp_cmd = Command(None, Eject (cd, Some true)) in
-          qmp_write device.frontend.domid qmp_cmd
+          qmp_send_cmd domid Qmp.(Eject(cd, Some true)) |> ignore
         else
           try
-            let c = Qmp_protocol.connect (qmp_libxl_path device.frontend.domid) in
+            let as_msg cmd = Qmp.(Success(Some __LOC__, cmd)) in
+            let fd_cd      = Unix.openfile params [ Unix.O_RDONLY ] 0o640 in
```

Comment:
Would it be better to use `Unixext.with_file` instead of `openfile [...] finally [...] close`?

Reply:
`Xapi_stdext_unix` is a large library that we are not yet using - @mseri would prefer not to add it.